Automattic/wp-calypso
# Pricing Grid: Implement Responsive

#### Proposed Changes

* This PR is one among several stacked PRs that implements the new pricing grid design. In https://github.com/Automattic/wp-calypso/pull/71372, we introduced a new signup flow /start/onboarding-2023-pricing-grid that is only available in dev and wpcalypso environments.
* The design for the pricing grid can be found in 2e991-pb. Please keep the Figma design open in a separate tab.
* The parent PR is https://github.com/Automattic/wp-calypso/pull/71516.
* This PR implements the responsive designs for the pricing grid. More details:
    * Refactor code to separate out desktop, tablet, and mobile views.
    * As given in Figma, implement desktop grids of 1320px and 1480px; tablet grid of 860px; and mobile grid of 375px.
    * Cleanup of unused code in several places.


https://user-images.githubusercontent.com/1269602/209778206-153e343c-8226-4301-be68-4314c9a885da.mov



#### What Not To Test

* Non-en. The first launch will be `en` only.

#### Testing Instructions



*  Navigate to `/start/onboarding-2023-pricing-grid`.
* Verify responsive web works correctly in all viewport sizes. In Desktop, Figma has two configurations - 1320px and 1480px, verify that you see each variation. For the former you'd need a screen width of atleast 1340px and for the latter, a width of at least 1500px. (I added an extra 20px so that the full width of the pricing grid still leaves some 10px margin on the left and right extremes instead of end-to-end wrapping the screen).
In Tablet,, Figma has a 860px wide configuration. Verify that you see the tablet configuration for less than 1340px screen width.
Verify that you see mobile configuration for less than 880px screen width. 
* Verify that the screen UI exactly matches Figma - paddings, margins, font sizes in different configurations, etc.
* Importantly, also verify that the main signup flow at `/start` is unaffected by these changes, since it touches a few SCSS files shared by the main signup flow.
